### PR TITLE
Port extractcode to Python 3 #295

### PR DESCRIPTION
**Pull request by Abhishek-Dev09**
The objective of PR is to ensure that all the tests of extractcode pass on both Python 2 & 3 by use of conffest plugin.

Signed-off-by: Abhishek Kumar <Abhishek.kasyap09@gmail.com>

### CODE REVIEW

File: src/commoncode/fileutils.py [hunk 1]
```diff
@@ -185,6 +185,17 @@ def get_temp_dir(base_dir=scancode_temp_dir, prefix=''):
...
+    """
+    Return the `pth` path string either as encoded bytes if on Linux and using
+    Python 2 or as a unicode/text otherwise.
+    """
+    if on_linux and py2 and isinstance(pth, compat.unicode):
+        return fsencode(pth)
+    if not isinstance(pth, compat.unicode):
+        return fsdecode(pth)
+    return pth
+
```
Comment (pombredanne) [inline src/commoncode/fileutils.py]: @Abhishek-Dev09 This is wrong IMHO
We should instead have something like this:
```
    if on_linux and py2:
        if not isinstance(pth, compat.bytes):
            pth = fsencode(pth)
        return path
    if not isinstance(pth, compat.unicode):
        return fsdecode(pth)
    return pth

```

Comment (Abhishek-Dev09) [reply src/commoncode/fileutils.py]: This changes have been made by you only.

Comment (pombredanne) [reply src/commoncode/fileutils.py]: Sure, that does not mean that you should not fix it.

Comment (Abhishek-Dev09) [reply src/commoncode/fileutils.py]: i have fixed

**Comment by pombredanne** [comment]
We are now green!
at last... There is a test not completed yet on the slower Appveyor But I am going to merge anyway.

File: src/commoncode/fileutils.py (hunk 1 above)
Comment (pombredanne) [reply src/commoncode/fileutils.py]: You can never hurt me!

**Pull request merged**